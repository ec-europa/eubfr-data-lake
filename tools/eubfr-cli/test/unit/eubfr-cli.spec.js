/**
 * @jest-environment node
 */

const path = require('path');
const { promisify } = require('util');
const { exec } = require('child_process');

const execute = promisify(exec);

// Execute from project root to make use of workspace's symlinking
const cwd = path.resolve(`${__dirname}../../../../../`);

describe('The EUBFR CLI: help menu', () => {
  test('is displayed if no arguments or options are passed', async () => {
    const command = `npx eubfr-cli`;
    const result = await execute(command, { cwd });
    expect(result.stdout).toMatchSnapshot();
  });

  test('is displayed on -h or --help flags', async () => {
    const command = `npx eubfr-cli`;
    const result = await execute(command, { cwd });
    expect(result.stdout).toMatchSnapshot();
  });
});

describe('The EUBFR CLI: upload command', () => {
  test('has a help menu on -h or --help', async () => {
    const command = `npx eubfr-cli upload -h`;
    const result = await execute(command, { cwd });
    expect(result.stdout).toMatchSnapshot();
  });

  test('requires `.content` folder if no options specified', async () => {
    const command = `npx eubfr-cli upload`;
    const result = execute(command, { cwd });
    expect(result).rejects.toThrow();
  });

  test('requires a producer when a file is specified', async () => {
    const command = `npx eubfr-cli upload foo`;
    const result = execute(command, { cwd });
    expect(result).rejects.toThrow('error: Missing required input parameters');
  });

  test('requires a correct value for the producer, even if the -p flag is used', async () => {
    const command = `npx eubfr-cli upload foo -p`;
    const result = execute(command, { cwd });
    expect(result).rejects.toThrow('error: Missing required input parameters');
  });

  test('requires a valid producer', async () => {
    const command = `npx eubfr-cli upload foo -p bar`;
    const result = await execute(command, { cwd });
    expect(result.stderr).toEqual(
      expect.stringMatching("Couldn't find credentials for producer")
    );
  });
});
