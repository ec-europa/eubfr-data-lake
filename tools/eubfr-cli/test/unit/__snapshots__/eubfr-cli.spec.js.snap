// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The EUBFR CLI: help menu is displayed if no arguments or options are passed 1`] = `
"
  Usage:  [command] [option]

  Options:

    -V, --version                output the version number
    -h, --help                   output usage information

  Commands:

    upload [options] [files...]  Uploads content to the data lake.
    show [options] [file]        Displays files of a given producer.
    delete [options] [files...]  Deletes files by computed_key field.
"
`;

exports[`The EUBFR CLI: help menu is displayed on -h or --help flags 1`] = `
"
  Usage:  [command] [option]

  Options:

    -V, --version                output the version number
    -h, --help                   output usage information

  Commands:

    upload [options] [files...]  Uploads content to the data lake.
    show [options] [file]        Displays files of a given producer.
    delete [options] [files...]  Deletes files by computed_key field.
"
`;

exports[`The EUBFR CLI: upload command has a help menu on -h or --help 1`] = `
"
  Usage: upload [options] [files...]

  Uploads content to the data lake.

  Options:

    -p, --producer [producer]  Producer's name.
    -h, --help                 output usage information
"
`;
